be/src/fs/fs.h

```diff
@@ -85,12 +85,16 @@ struct SequentialFileOptions {
     // Don't cache remote file locally on read requests.
     // This options can be ignored if the underlying filesystem does not support local cache.
     bool skip_fill_local_cache = false;
+    // Specify different buffer size for different read scenarios
+    int64_t buffer_size = -1;
 };
 
 struct RandomAccessFileOptions {
     // Don't cache remote file locally on read requests.
     // This options can be ignored if the underlying filesystem does not support local cache.
     bool skip_fill_local_cache = false;
+    // Specify different buffer size for different read scenarios
+    int64_t buffer_size = -1;
```

Comment:
How about add a field of type `map<std::string, std::string> properties` for passing custom properties? cc @kevincai @wyb

Reply:
IMHO, there is a high probability that we will need to add other custom properties in the future, and using a map here allows for better extensibility.

Reply:
> IMHO, there is a high probability that we will need to add other custom properties in the future, and using a map here allows for better extensibility.

That was my concern too. I will refactor it.

Reply:
> > IMHO, there is a high probability that we will need to add other custom properties in the future, and using a map here allows for better extensibility.
> 
> That was my concern too. I will refactor it.

Perhaps we could go one step closer and encapsulate `map<std::string, std::string>` into a utility class `Properties` with some convenient methods for type conversion, like the following:
```
class Properties {
public:
    std::optional<bool> get_boolean_property(std::string key) { ... }
    std::optional<int> get_int_property(std::string key) { ... }
    ...
private:
    std::map<std::string， std::string> _map;
};
```